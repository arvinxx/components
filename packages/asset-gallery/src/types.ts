import type { ReactNode, CSSProperties } from 'react';

export interface Asset {
  title: string;
  /**
   * 图片 url
   */
  url: string;
  description?: string;

  dark?: boolean;
  /**
   * 如果需要反色的话
   * 可以设置反色的颜色
   */
  darkBackground?: string;
  /**
   * 允许添加间距
   */
  padding?: number | string;
  /**
   * 如果有 sketch
   */
  sketch?: string;
}

export type AssetList = Asset[];

export interface AssetGalleryData {
  data: AssetList;
}

export interface ImageEntryYML extends Omit<Asset, 'url' | 'title'> {
  标题: string;
  title?: string;
  描述?: string;
  反色?: boolean;
  反色背景色?: string;
  链接: string;
  url?: string;
}

export type AssetGalleryYML = ImageEntryYML[];

export type LayoutType = 'grid' | 'masonry';

export interface AssetGalleryProps {
  /**
   * 数据清单
   */
  data: AssetList | string;
  /**
   * 暗色背景下默认的背景色
   * @default #1890ff
   */
  darkBackground?: string;
  /**
   * 使用的布局方式: 瀑布流与网格
   * @default masonry
   */
  layout?: LayoutType;
  /**
   * 布局切换时的回调
   * @param layout
   */
  onLayoutChange?: (layout: LayoutType) => void;

  /**
   * 显示logo
   */
  logo?: ReactNode;

  /**
   * 列数
   */
  columns?: number;
  /**
   * 列数改变时的回调
   * @param columns
   */
  onColumnsChange?: (columns: number) => void;

  /**
   * 显示滑杆
   */
  showSlider?: boolean;

  /**
   * 样式
   */
  style?: CSSProperties;
}
