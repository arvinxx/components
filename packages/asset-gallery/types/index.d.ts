declare class ClipboardItem {
  constructor(data: Record<string, Blob>);
}

interface Clipboard {
  read?: () => Promise<ClipboardItem[]>;

  write?: (items: ClipboardItem[]) => Promise<void>;
}
