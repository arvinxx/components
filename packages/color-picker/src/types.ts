import type { CSSProperties } from 'react';
import type React from 'react';

export interface HSLColor {
  h: number;
  l: number;
  s: number;
}

export interface HSVColor {
  h: number;
  s: number;
  v: number;
}

export interface RGBColor {
  b: number;
  g: number;
  r: number;
}

interface Classes<T> {
  default: Partial<T>;
  [scope: string]: Partial<T>;
}

export type Color = string | HSLColor | RGBColor;

export interface ColorResult {
  hex: string;
  hsl: HSLColor;
  rgb: RGBColor;
  source?: string;
}

export type ColorChangeHandler = (
  color: ColorResult,
  event: React.ChangeEvent<HTMLInputElement>,
) => void;

export interface SketchPickerStylesProps {
  picker: CSSProperties;
  saturation: CSSProperties;
  Saturation: CSSProperties;
  controls: CSSProperties;
  sliders: CSSProperties;
  color: CSSProperties;
  activeColor: CSSProperties;
  hue: CSSProperties;
  Hue: CSSProperties;
  alpha: CSSProperties;
  Alpha: CSSProperties;
}

export type PresetColor = { color: string; title: string } | string;

export interface ColorPickerProps {
  color?: Color | undefined;
  className?: string | undefined;

  onChange?: ColorChangeHandler;
  onChangeComplete?: ColorChangeHandler | undefined;

  disableAlpha?: boolean | undefined;
  presetColors?: PresetColor[] | undefined;
  width?: string | undefined;
  styles?: Partial<Classes<SketchPickerStylesProps>> | undefined;
  onSwatchHover?: (color: ColorResult, event: MouseEvent) => void;
}
