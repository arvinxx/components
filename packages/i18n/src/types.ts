export type LocalesType = 'en-US' | 'zh-CN';

export type LocaleMessageMaps<T = Record<string, string>> = {
  'zh-CN': T;
  'en-US'?: T;
};

export type { PrimitiveType } from 'intl-messageformat';
