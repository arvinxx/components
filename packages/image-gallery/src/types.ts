interface ImageEntry {
  title: string;
  description: string;

  dark?: boolean;
  url: string;
}

export type ImageList = ImageEntry[];
