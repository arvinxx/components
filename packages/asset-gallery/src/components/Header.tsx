import { Slider, Space } from 'antd';
import type { FC } from 'react';
import React from 'react';
import type { AssetGalleryProps } from '../types';
import LayoutButton from './LayoutButton';

type HeaderProps = Pick<
  AssetGalleryProps,
  | 'logo'
  | 'columns'
  | 'layout'
  | 'onColumnsChange'
  | 'onLayoutChange'
  | 'showSlider'
>;

const Header: FC<HeaderProps> = ({
  logo,
  onColumnsChange,
  columns,
  layout,
  onLayoutChange,
  showSlider,
}) => {
  const isMasonry = layout === 'masonry';
  return (
    <div
      style={{
        padding: '0 8px',
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
      }}
    >
      <div>
        {typeof logo === 'string' ? (
          <img src={logo} width={40} alt="logo" />
        ) : (
          logo
        )}
      </div>
      <Space>
        <Space size={0}>
          <LayoutButton
            icon={{
              default:
                'https://gw.alipayobjects.com/zos/antfincdn/a72YnCXMKW/masonry-default.svg',
              active:
                'https://gw.alipayobjects.com/zos/antfincdn/gACoIFWH4G/masonry.svg',
            }}
            onClick={() => {
              onLayoutChange('masonry');
            }}
            isActive={isMasonry}
            tooltip={'瀑布流布局'}
            testId={'masonry-icon'}
          />
          <LayoutButton
            icon={{
              default:
                'https://gw.alipayobjects.com/zos/antfincdn/sLiHETRbAt/grid.svg',
              active:
                'https://gw.alipayobjects.com/zos/antfincdn/lr6NsCRih%24/grid-active.svg',
            }}
            onClick={() => {
              onLayoutChange('grid');
            }}
            isActive={!isMasonry}
            tooltip={'网格布局'}
            testId={'grid-icon'}
          />
        </Space>
        {showSlider ? (
          <Slider
            style={{ width: 100 }}
            value={-columns}
            tooltipVisible={false}
            onChange={
              /* istanbul ignore next */
              (v) => onColumnsChange(-v)
            }
            max={-1}
            min={-4}
          />
        ) : null}
      </Space>
    </div>
  );
};
export default Header;
