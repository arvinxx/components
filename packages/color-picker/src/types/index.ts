export * from './swatch';
export * from './component';
export * from './store';
