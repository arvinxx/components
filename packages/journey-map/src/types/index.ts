export * from './type';
export * from './yml';
