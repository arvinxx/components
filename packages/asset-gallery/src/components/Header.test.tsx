import React, { useState } from 'react';
import { render, act } from '@testing-library/react';

import Header from './Header';
import { LayoutType } from '../types';

describe('Header', () => {
  it('默认', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
  it('有 logo', () => {
    const { container } = render(<Header logo={'123'} showSlider />);
    expect(container).toMatchSnapshot();
  });
  it('切换布局', async () => {
    const App = () => {
      const [layout, setLayout] = useState<LayoutType>('grid');
      return (
        <Header
          logo={'123'}
          layout={layout}
          onLayoutChange={setLayout}
          showSlider
        />
      );
    };
    act(() => {});
    const { findByTestId } = render(<App />);
    const masonryBtn = await findByTestId('masonry-icon');
    const gridBtn = await findByTestId('grid-icon');

    masonryBtn.click();
    gridBtn.click();
  });
});
