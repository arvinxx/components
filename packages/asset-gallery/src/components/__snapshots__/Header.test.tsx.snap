// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 有 logo 1`] = `
<div>
  <div
    style="padding: 0px 8px; display: flex; justify-content: space-between; align-items: center;"
  >
    <div>
      <img
        alt="logo"
        src="123"
        width="40"
      />
    </div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 0px;"
          >
            <button
              class="ant-btn ant-btn-link ant-btn-icon-only"
              data-testid="masonry-icon"
              type="button"
            >
              <img
                alt="瀑布流布局"
                src="https://gw.alipayobjects.com/zos/antfincdn/Hj%26JKvBCCl/masonry-default.svg"
                style="cursor: pointer;"
              />
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-link ant-btn-icon-only"
              data-testid="grid-icon"
              type="button"
            >
              <img
                alt="网格布局"
                src="https://gw.alipayobjects.com/zos/antfincdn/lr6NsCRih%24/grid-active.svg"
                style="cursor: pointer;"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-slider"
          style="width: 100px;"
        >
          <div
            class="ant-slider-rail"
          />
          <div
            class="ant-slider-track"
            style="left: 0%; width: 133.33333333333331%;"
          />
          <div
            class="ant-slider-step"
          />
          <div
            aria-disabled="false"
            aria-valuemax="-1"
            aria-valuemin="-4"
            aria-valuenow="0"
            class="ant-slider-handle"
            role="slider"
            style="left: 133.33333333333331%; transform: translateX(-50%);"
            tabindex="0"
          />
          <div
            class="ant-slider-mark"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header 默认 1`] = `
<div>
  <div
    style="padding: 0px 8px; display: flex; justify-content: space-between; align-items: center;"
  >
    <div />
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 0px;"
          >
            <button
              class="ant-btn ant-btn-link ant-btn-icon-only"
              data-testid="masonry-icon"
              type="button"
            >
              <img
                alt="瀑布流布局"
                src="https://gw.alipayobjects.com/zos/antfincdn/Hj%26JKvBCCl/masonry-default.svg"
                style="cursor: pointer;"
              />
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-link ant-btn-icon-only"
              data-testid="grid-icon"
              type="button"
            >
              <img
                alt="网格布局"
                src="https://gw.alipayobjects.com/zos/antfincdn/lr6NsCRih%24/grid-active.svg"
                style="cursor: pointer;"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
